design/cluster-scope-resource-filter.md

```diff
@@ -0,0 +1,572 @@
+	--include-cluster-resources=true
+```
+
+This means backup Deployment, Service, Endpoint, Pod and ReplicaSet resources in default and kube-system namespaces, and related cluster resources, plus all StorageClass cluster resources.
+``` bash
+velero backup create <backup-name>
+	--include-namespaced-resources=Deployment,Service,Endpoint,Pod,ReplicaSet
+	--include-namespaces=default,kube-system
+	--include-cluster-scope-resources=StorageClass
+	--include-cluster-resources=true
+```
+
+This means backup Deployment, Service, Endpoint, Pod and ReplicaSet resources in default and kube-system namespaces, and related cluster resources, plus all cluster scope resources except StorageClass type resources.
+``` bash
+velero backup create <backup-name>
+	--include-namespaced-resources=Deployment,Service,Endpoint,Pod,ReplicaSet
+	--include-namespaces=default,kube-system
+	--exclude-cluster-scope-resources=StorageClass
+	--include-cluster-resources=true
+```
+
+#### some namespaced resources + all cluster resources
+This means backup all resources in namespace in default, kube-system, and all cluster resources.
+``` bash
+velero backup create <backup-name>
+	--include-namespaces=default,kube-system
+	--include-cluster-scope-resources=*
+	--include-cluster-resources=true
+```
+
+This means backup Deployment, Service, Endpoint, Pod and ReplicaSet resources in all namespaces, and all cluster resources.
+``` bash
+velero backup create <backup-name>
+	--include-namespaced-resources=Deployment,Service,Endpoint,Pod,ReplicaSet
+	--include-cluster-resources=true
+```
+
+This means backup Deployment, Service, Endpoint, Pod and ReplicaSet resources in default and kube-system namespaces, and all cluster resources.
+``` bash
+velero backup create <backup-name>
+	--include-namespaces=default,kube-system
+	--include-namespaced-resources=Deployment,Service,Endpoint,Pod,ReplicaSet
+	--include-cluster-resources=true
+```
+
+#### all namespaced resources + no cluster resources
+The following commands all mean backup all namespace scope resources and no cluster resources.
+``` bash
+velero backup create <backup-name>
+	--include-namespaces=*
+	--include-cluster-resources=false
+```
+
+``` bash
+velero backup create <backup-name>
+	--include-namespaced-resources=*
+	--include-cluster-resources=false
+```
+
+``` bash
+velero backup create <backup-name>
+	--include-namespaces=*
+	--include-namespaced-resources=*
+	--include-cluster-resources=false
+```
+
+``` bash
+velero backup create <backup-name>
+	--include-cluster-resources=false
+```
+
+#### all namespaced resources + only related cluster resources
+The following commands all mean backup all resources in all namespaces, and related cluster resources.
+``` bash
+velero backup create <backup-name>
+	--include-namespaces=*
+```
+
+``` bash
+velero backup create <backup-name>
+	--include-namespaced-resources=*
+```
+
+``` bash
+velero backup create <backup-name>
+	--include-namespaces=*
+	--include-namespaced-resources=*
```

Comment:
With the above, you'll also need --include-cluster-scope-resources="*" If you remove the line above, then it's not needed as we'll be using the old includes/excludes rules

Reply:
This case includes related cluster sources, not all cluster resources.

Reply:
Agreed. See above comment for why we still need to make sure that we're not excluding all cluster resources in the resource list.